MapboxNavigation/RouteMapViewController.swift

```diff
@@ -287,6 +288,32 @@ class RouteMapViewController: UIViewController {
             mapView.removeArrow()
         }
     }
+
+    func updateCameraAltitude(for routeProgress: RouteProgress) {
+        guard mapView.tracksUserCourse else { return } //only adjust when we are actively tracking user course
+        
+        let zoomOutAltitude = NavigationMapView.zoomedOutManeuverAltitude
+        let defaultAltitude = NavigationMapView.defaultAltitude
+        let isLongRoad = routeProgress.distanceRemaining >= NavigationMapView.longManeuverDistance
+        
+        
+        let currentStepIsMotorway = currentStep.isMotorway
+        let nextStepIsMotorway = upComingStep?.isMotorway ?? false
+        let isExiting = currentStepIsMotorway && !nextStepIsMotorway //are we exiting a motorway?
+        let notOnMotorway = !currentStepIsMotorway && !nextStepIsMotorway //are we not on a motorway?
+        
+        if (notOnMotorway || isExiting) { //if we're exiting or not on a motorway, we should be zoomed in.
+            return setCamera(altitude: defaultAltitude)
+        }
+        if currentStepIsMotorway, isLongRoad { //otherwise, we should be zoomed out if we're on motorway and step distance is long enough
```

Comment:
Should this be an else if?

Reply:
It won't affect the logic of anything, but it certainly can be.